src/main/java/uk/gov/hmcts/reform/slc/services/steps/getpdf/FileNameGenerator.java

```diff
@@ -1,16 +1,16 @@
 package uk.gov.hmcts.reform.slc.services.steps.getpdf;
 
+import uk.gov.hmcts.reform.slc.model.Letter;
+
 import static org.apache.commons.codec.digest.DigestUtils.md5Hex;
 
 public final class FileNameGenerator {
 
     public static String generateFor(
-        String letterType,
-        String jurisdiction,
-        byte[] content,
+        Letter letter,
         String extension
     ) {
-        return letterType + "-" + jurisdiction + "-" + md5Hex(content) + "." + extension;
+        return letter.type + "-" + letter.service + "-" + md5Hex(letter.messageId) + "." + extension;
```

Comment:
md5 of md5?

Reply:
Good catch didn't realised thought it's a string.Will update that to just have message id